### DISCUSSION TITLE
Using Transitions between pages?

### DISCUSSION THREAD

**Discussion by saltnpixels** [vercel/next.js#18143] [Help]
I'm new to next.js let alone react.

I started making a page and wanted to use CSS Transitions Group to show hide parts of the page based on clicking some sub menu. (https://reactcommunity.org/react-transition-group/)

I realized that I can't transition from one next.js page to another. So I made one page that catches everything using [...slug].js. This works. I shoow and hide stuff based on the url.
And then I had to get the next.js router component and basically make a subrouter that transitions between stuff for this one page.

Am I reinventing the wheel? Is this how I am suppose to do this if i want to transition between pages? 
Im new! any help would be greatly appreciated!

**Comment by jamesmosier** [comment]
Hi @saltnpixels! Have you checked out [with-next-page-transition](https://github.com/vercel/next.js/tree/canary/examples/with-next-page-transitions)? It might help for some inspiration! Especially the [_app.js](https://github.com/vercel/next.js/blob/canary/examples/with-next-page-transitions/pages/_app.js).
